Assets/Scripts/UI/Pickup Pup/DogProfile/DogAdoptProfile.cs

```diff
@@ -13,22 +13,41 @@ public class DogAdoptProfile : DogProfile
     [SerializeField]
     Button adoptButton;
 
+    Color defaultPriceColor;
+    Color overpricedColor = Color.red;
+
+    #region MonoBehaviourExtended Overrides
+
     protected override void setReferences()
     {
         base.setReferences();
-        rehomeButton.interactable = false;
-        collarSlot.interactable = false;
+        iconsObject.SetActive(false);
+        defaultPriceColor = priceText.color;
     }
 
+    #endregion
+
+    #region DogProfile Overrides
+
     public override void SetProfile(Dog dog)
     {
         base.SetProfile(dog);
         priceText.text = dogInfo.CostToAdoptStr;
-    }
 
-    public void TryAdopt()
-    {
-        game.TryAdoptDog(dogInfo);
+        if(!game)
+        {
+            fetchReferences();
```

Comment:
Because `fetchReferences` is automatically called in `Start`, it's better not to call it elsewhere. However, you can call `checkReferences` to call `fetchReferences` if it has not already run